apps/keyboard/js/settings/layout_item_list_view.js

```diff
@@ -60,13 +60,26 @@ var Deferred = function() {
   }.bind(this));
 };
 
+// DialogInfo holds the information of one showDialog() call so we could
+// work on that request when we actually do so.
+var DialogInfo = function(args) {
+  this.args = args;
+  this.deferred = new Deferred();
+};
+
+// This is a wrapper that wraps <gaia-confirm> into our view management.
+// The reason we did this is because we need to defer the toast from showing,
+// if our containing view is not visible.
+// We also want to queue the showDialog requests.
 var ConfirmationDialogBaseView = function() {
   BaseView.apply(this);
 
-  this.deferred = null;
-};
+  this._isVisible = false;
```

Comment:
I took some time to understand the purpose of these variables, maybe we could add some comments here to describe them.